addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -364,3 +364,35 @@ def test_service_to_purchase_multi_company(self):
         # FIXME: same sudo issue as above
         order2.sudo().with_company(company_2).action_confirm()
         self.assertTrue(order2.purchase_order_count)
+
+    def test_service_to_purchase_branch_tax_propagation(self):
+        """
+            Test that taxes are passed from the SO to the PO for service products in branch companies
+        """
+        branch = self.env['res.company'].create({
+            'name': "Branch Company",
+            'parent_id': self.env.company.id,
+        })
+        service_product = self.env['product.product'].with_company(branch).create({
+            'name': "Branch Out-sourced Service",
+            'standard_price': 200.0,
+            'type': 'service',
+            'invoice_policy': 'delivery',
+            'taxes_id': self.company_data['default_tax_sale'],
```

Comment:
Let's make the supplier tax explicit so we can quickly understand the assert
```suggestion
            'taxes_id': self.company_data['default_tax_sale'],
            'supplier_taxes_id': self.company_data['default_tax_purchase'],
```